### ISSUE TITLE
Moving category into a new one manually in the Back-Office

### ISSUE THREAD

**Issue by theozzz** [Smile-SA/elasticsuite#337]
Hello all,

We encounter a new issue when trying to move a category into a new one manually in the BO.

Please check the exception below with the full stack trace : 

```
==> var/log/exception.log <==
[2017-02-28 10:56:48] report.CRITICAL: exception 'Exception' with message 'Recoverable Error: Object of class Smile\ElasticsuiteVirtualCategory\Model\Rule could not be converted to string in /var/www/avatacar/current/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php on line 228' in /var/www/avatacar/current/vendor/magento/framework/App/ErrorHandler.php:61
Stack trace:
#0 [internal function]: Magento\Framework\App\ErrorHandler->handler(4096, 'Object of class...', '/var/www/avatac...', 228, Array)
#1 /var/www/avatacar/current/vendor/magento/zendframework1/library/Zend/Db/Statement/Pdo.php(228): PDOStatement->execute(Array)
#2 /var/www/avatacar/current/vendor/magento/framework/DB/Statement/Pdo/Mysql.php(95): Zend_Db_Statement_Pdo->_execute(Array)
#3 /var/www/avatacar/current/vendor/magento/zendframework1/library/Zend/Db/Statement.php(323): Magento\Framework\DB\Statement\Pdo\Mysql->_execute(Array)
#4 /var/www/avatacar/current/vendor/magento/zendframework1/library/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#5 /var/www/avatacar/current/vendor/magento/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(238): Zend_Db_Adapter_Abstract->query('INSERT INTO `ca...', Array)
#6 /var/www/avatacar/current/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php(459): Zend_Db_Adapter_Pdo_Abstract->query('INSERT INTO `ca...', Array)
#7 /var/www/avatacar/current/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php(514): Magento\Framework\DB\Adapter\Pdo\Mysql->_query('INSERT INTO `ca...', Array)
#8 /var/www/avatacar/current/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php(1860): Magento\Framework\DB\Adapter\Pdo\Mysql->query('INSERT INTO `ca...', Array)
#9 /var/www/avatacar/current/vendor/magento/module-catalog/Model/Indexer/Category/Flat/Action/Rows.php(98): Magento\Framework\DB\Adapter\Pdo\Mysql->insertOnDuplicate('catalog_categor...', Array, Array)
#10 /var/www/avatacar/current/vendor/magento/module-catalog/Model/Indexer/Category/Flat.php(65): Magento\Catalog\Model\Indexer\Category\Flat\Action\Rows->reindex(Array)
#11 /var/www/avatacar/current/vendor/magento/module-catalog/Model/Indexer/Category/Flat.php(88): Magento\Catalog\Model\Indexer\Category\Flat->execute(Array)
#12 /var/www/avatacar/current/vendor/magento/module-indexer/Model/Indexer.php(445): Magento\Catalog\Model\Indexer\Category\Flat->executeList(Array)
#13 /var/www/avatacar/current/var/generation/Magento/Indexer/Model/Indexer/Interceptor.php(349): Magento\Indexer\Model\Indexer->reindexList(Array)
#14 /var/www/avatacar/current/vendor/magento/module-catalog/Model/Category.php(438): Magento\Indexer\Model\Indexer\Interceptor->reindexList(Array)
#15 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Catalog\Model\Category->move('14', '0')
#16 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Catalog\Model\Category\Interceptor->___callParent('move', Array)
#17 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'move', Object(Magento\Catalog\Model\Category\Interceptor), Array, 'categoryChange')
#18 /var/www/avatacar/current/vendor/magento/module-catalog-permissions/Model/Indexer/Plugin/Category.php(86): Magento\Catalog\Model\Category\Interceptor->Magento\Framework\Interception\{closure}('14', '0')
#19 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(142): Magento\CatalogPermissions\Model\Indexer\Plugin\Category->aroundMove(Object(Magento\Catalog\Model\Category\Interceptor), Object(Closure), '14', '0')
#20 /var/www/avatacar/current/var/generation/Magento/Catalog/Model/Category/Interceptor.php(78): Magento\Catalog\Model\Category\Interceptor->___callPlugins('move', Array, Array)
#21 /var/www/avatacar/current/vendor/magento/module-catalog/Controller/Adminhtml/Category/Move.php(69): Magento\Catalog\Model\Category\Interceptor->move('14', '0')
#22 /var/www/avatacar/current/var/generation/Magento/Catalog/Controller/Adminhtml/Category/Move/Interceptor.php(24): Magento\Catalog\Controller\Adminhtml\Category\Move->execute()
#23 /var/www/avatacar/current/vendor/magento/framework/App/Action/Action.php(102): Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor->execute()
#24 /var/www/avatacar/current/vendor/magento/module-backend/App/AbstractAction.php(226): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#25 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Backend\App\AbstractAction->dispatch(Object(Magento\Framework\App\Request\Http))
#26 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor->___callParent('dispatch', Array)
#27 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor), Array, 'adminAuthentica...')
#28 /var/www/avatacar/current/vendor/magento/module-backend/App/Action/Plugin/Authentication.php(143): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#29 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch(Object(Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#30 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor), Array, 'log')
#31 /var/www/avatacar/current/vendor/magento/module-logging/App/Action/Plugin/Log.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#32 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Logging\App\Action\Plugin\Log->aroundDispatch(Object(Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#33 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor), Array, 'adminMassaction...')
#34 /var/www/avatacar/current/vendor/magento/module-backend/App/Action/Plugin/MassactionKey.php(33): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#35 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Backend\App\Action\Plugin\MassactionKey->aroundDispatch(Object(Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#36 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Catalog...', 'dispatch', Object(Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor), Array, 'mgtDeveloperToo...')
#37 /var/www/avatacar/current/vendor/mgtcommerce/module-mgtdevelopertoolbar/src/app/code/Mgt/DeveloperToolbar/Model/App/FrontController/ToolbarPlugin.php(64): Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#38 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(142): Mgt\DeveloperToolbar\Model\App\FrontController\ToolbarPlugin->aroundDispatch(Object(Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#39 /var/www/avatacar/current/var/generation/Magento/Catalog/Controller/Adminhtml/Category/Move/Interceptor.php(39): Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor->___callPlugins('dispatch', Array, Array)
#40 /var/www/avatacar/current/vendor/magento/framework/App/FrontController.php(55): Magento\Catalog\Controller\Adminhtml\Category\Move\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#41 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#42 /var/www/avatacar/current/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#43 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#44 /var/www/avatacar/current/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#45 /var/www/avatacar/current/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#46 /var/www/avatacar/current/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#47 /var/www/avatacar/current/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#48 /var/www/avatacar/current/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#49 /var/www/avata
```

FYI : 
Magento EE: 2.1.2
Smile ElasticSuite: 2.3.1

Kind Regards,

Théo

**Comment by romainruaud** [comment]
Confirmed.

We'll provide a fix for this one on the next minor release of ElasticSuite.

Thank you for reporting it.

**Comment by romainruaud** [comment]
Fixed by #348 .

Please consider using 2.3.x branch directly if you wanna benefit from this fix fast.

Otherwise, you can wait to the next minor relase of the module.

Best regards,
